File: codigo/soluciones/kat/buttonbashing.cpp
#include <bits/stdc++.h>
using namespace std;
signed main() {
    cin.sync_with_stdio(false);
    cin.tie(nullptr);
    int T;
    cin >> T;
    while (T--) {
        int n, t;
        cin >> n >> t;
        vector<int> buttons(n);
        for (auto &it : buttons) {
            cin >> it;
        }
        vector<int> ans(3601, -1);
        int aux = 0;
        queue<int> q;
        q.push(0);
        q.push(-1);
        while (q.size() > 1) {
            int i = q.front();
            q.pop();
            if (i == -1) {
                ++aux;
                q.push(-1);
                continue;
            }
            if (ans[i] != -1) continue;
            ans[i] = aux;
            for (auto &it : buttons) {
                if (ans[max(0, min(3600, i + it))] != -1) continue;
                q.push(max(0, min(3600, i + it)));
            }
        }
        aux = t;
        while (ans[aux] == -1) {
            ++aux;
        }
        cout << ans[aux] << ' ' << aux - t << '\n';
    }
    cout.flush();
    return 0;
}

File: codigo/soluciones/kat/buttonbashing.py
T = int(input())
for _ in range(T):
    n, t = list(map(int, input().split()))
    buttons = list(map(int, input().split()))
    ans = [-1 for _ in range(3601)]
    aux = 0
    q = []
    q.append(0)
    q.append(-1)
    while len(q) > 1:
        i = q.pop(0)
        if i == -1:
            aux += 1
            q.append(-1)
            continue
        if ans[i] != -1:
            continue
        ans[i] = aux
        for button in buttons:
            if ans[max(0, min(3600, i + button))] != -1:
                continue
            q.append(max(0, min(3600, i + button)))
    aux = t
    while ans[aux] == -1:
        aux += 1
    print(ans[aux], aux - t)